hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```diff
@@ -15,37 +66,76 @@ public SerializableVector(IResetableComparableFactory factory, int recordSize){
      * @param frameSize
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int frameSize, int recordSize){
-
+    public SerializableVector(int recordSize, int frameSize){
+        frames = new ArrayList<>();
+        init(recordSize);
+        this.frameSize = frameSize;
+        recordPerFrame = frameSize / recordSize;
+        ctx = new FrameManager(frameSize);
```

Comment:
The ctx is always passed by the caller. Otherwise, we can not have a centralized memory management.